### PR TITLE
Task/add holiday service and controller functionality

### PR DESCRIPTION
**Pull request by Jamie-Unosquare**
Currently left out Create functionality as it needs to touch a few different classes

### CODE REVIEW

File: v2/AdminCore.Services/EventService.cs [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+      return DatabaseStatus.Failure;
+    }
+
+    public DatabaseStatus ApproveEvent(EventDto eventDto)
+    {
+      var existingEvent = _databaseContext.EventRepository.GetById(eventDto);
+      if (existingEvent != null)
+      {
+        Event eventToUpdate = existingEvent;
```
Comment (nmcmahon-unosquare) [inline v2/AdminCore.Services/EventService.cs]: refactor into method updateEventStatus

File: v2/AdminCore.Services/EventService.cs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+      return DatabaseStatus.Failure;
+    }
+
+    public DatabaseStatus CancelEvent(EventDto eventDto)
+    {
+      var existingEvent = _databaseContext.EventRepository.GetById(eventDto);
+      if (existingEvent != null)
+      {
+        Event eventToCancel = existingEvent;
```
Comment (nmcmahon-unosquare) [inline v2/AdminCore.Services/EventService.cs]: Same here - refactor

File: v2/AdminCore.Services/EventService.cs [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+      List<string> responses = new List<string>();
+      var existingEvent = _databaseContext.EventRepository.GetById(eventId);
+      if (existingEvent != null)
+      {
+        Event eventToReject = existingEvent;
+        eventToReject.EventStatus.EventStatusId = (int)EventStatuses.Rejected;
+        Employee employee = GetEmployeeFromEmployeeId(employeeId);
+        if (employee != null)
+        {
+          if (eventToReject.EventStatus.EventStatusId != (int)EventStatuses.Rejected)
```
Comment (nmcmahon-unosquare) [inline v2/AdminCore.Services/EventService.cs]: 3 levels of nested ifs - make line 136-151 a method and look at extracting your nested ifs into further, smaller methods

File: v2/AdminCore.Services/EventService.cs [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+      var eventsBetweenDates = _databaseContext.EventDatesRepository.Get(x => x.StartDate >= eventDateDto.StartDate
+                                                                              && x.EndDate <= eventDateDto.EndDate
+                                                                              && x.Event.EventTypeId ==
+                                                                              (int)EventTypes.AnnualLeave);
+      return _mapper.Map<List<EventDto>>(eventsBetweenDates);
+    }
+
+    public IList<EventDto> GetByEmployeeId(int employeeId)
+    {
```
Comment (eoinmcafee) [inline v2/AdminCore.Services/EventService.cs:71]: This logic should live in the employee service

**Comment by nmcmahon-unosquare** [comment]
Closing as all functionality is included in later PR

**Pull request closed without merging**